src/Servers/Kestrel/Core/src/CertificateLoader.cs

```diff
@@ -36,13 +36,19 @@ public static X509Certificate2 LoadFromStoreCert(string subject, string storeNam
                 {
                     store.Open(OpenFlags.ReadOnly);
                     storeCertificates = store.Certificates;
-                    var foundCertificates = storeCertificates.Find(X509FindType.FindBySubjectName, subject, !allowInvalid);
-                    foundCertificate = foundCertificates
+                    foreach (var certificate in storeCertificates.Find(X509FindType.FindBySubjectName, subject, !allowInvalid)
                         .OfType<X509Certificate2>()
                         .Where(IsCertificateAllowedForServerAuth)
                         .Where(DoesCertificateHaveAnAccessiblePrivateKey)
-                        .OrderByDescending(certificate => certificate.NotAfter)
-                        .FirstOrDefault();
+                        .OrderByDescending(certificate => certificate.NotAfter))
+                    {
+                        foundCertificate = certificate;
```

Comment:
If we never find an exact match, foundCertificate will now be the _last_ element in the collection we're iterating on, whereas before it was the first.

Reply:
good catch, to be updated